File: java-service/src/main/java/com/edu/platform/model/LessonTopic.java
package com.edu.platform.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "lesson_topics", schema = "academic")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class LessonTopic {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String topicTitle;
    
    @Column(length = 500)
    private String description;
    
    /** Entity relationship — Subject */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "subject_id", nullable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler", "department"})
    private Subject subject;
    
    /** Entity relationship — Term */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "term_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Term term;
    
    @Column(name = "week_number")
    private int weekNumber;
    
    @Column(name = "file_name", length = 255)
    private String fileName;
    
    @Column(name = "file_url", length = 500)
    private String fileUrl;
    
    @Column(name = "question_count")
    private Integer questionCount = 0;
    
    @Column(name = "is_aspirant_material", nullable = false)
    private boolean isAspirantMaterial = false;
    
    

    /**
     * Alias for topicTitle (used in VideoProcessingOrchestrator)
     */
    public String getTitle() {
        return this.topicTitle;
    }

    public void setTitle(String title) {
        this.topicTitle = title;
    }
}

File: python-service/app/models/lesson_topic.py
from sqlalchemy import Column, Integer, String, Boolean, ForeignKey
from sqlalchemy.orm import relationship
from app.models.base import Base


class LessonTopic(Base):
    """Lesson Topic model - matches Java LessonTopic entity"""
    __tablename__ = "lesson_topics"
    __table_args__ = {"schema": "academic"}

    id = Column(Integer, primary_key=True, index=True)
    topic_title = Column(String(255), nullable=False)
    description = Column(String(500))
    file_url = Column(String(500))
    file_name = Column(String(255))
    week_number = Column(Integer)
    question_count = Column(Integer, default=0)
    is_aspirant_material = Column(Boolean, default=False, nullable=False)
    
    # Foreign Keys
    subject_id = Column(Integer, ForeignKey("academic.subjects.id"), nullable=False)
    term_id = Column(Integer, ForeignKey("academic.terms.id"))
    
    # Relationships - ONLY the ones Java has
    subject = relationship("Subject", lazy="select", foreign_keys=[subject_id])
    term = relationship("Term", lazy="select", foreign_keys=[term_id])
    lesson_ai_results = relationship(
        "LessonAIResult",
        back_populates="lesson_topic",
        cascade="all, delete-orphan",
        lazy="select"
    )
    # DON'T add live_sessions - Java doesn't have @OneToMany for it